doc/sphinx/proofs/writing-proofs/inductive.rst

```diff
@@ -5,14 +5,14 @@ Reasoning with inductive types
 Applying constructors
 ---------------------
 
-.. tacn:: constructor @natural
-   :name: constructor
+.. tacn:: constructor {? @nat_or_var } {? with @bindings }
 
-   This tactic applies to a goal such that its conclusion is an inductive
-   type (say :g:`I`). The argument :token:`natural` must be less or equal to the
-   numbers of constructor(s) of :g:`I`. Let :n:`c__i` be the i-th
-   constructor of :g:`I`, then :g:`constructor i` is equivalent to
-   :n:`intros; apply c__i`.
+   .. todo PR: This really does an intros, peeling off foralls and -> ?
+      If @nat_or_var is not given, it just picks the right contructor?
+      (in which case, why would you ever bother to give @nat_or_var?).
```

Comment:
>This really does an intros, peeling off foralls and -> ?

Yes.

>If `@nat_or_var` is not given, it just picks the right contructor?

It picks one constructor that can be applied to the current goal, but it's not always the "right" one. So in some cases, you should be explicit on the constructor you want to apply.

E.g.:

```coq
Goal forall A B : Prop, B -> A \/ B.
constructor. (* does the same thing as constructor 1 / left, because it succeeds,
                but it makes the goal unprovable *)
(*
1 subgoal
A, B : Prop
H : B
______________________________________(1/1)
A
*)
Undo.
constructor 2. (* does the same thing as right *)
(*
1 subgoal
A, B : Prop
H : B
______________________________________(1/1)
B
*)

Reply:
Uh.  Backtracking may complicate the explanation of many, many tactics.  Any feel for how many tactics can produce multiple results with backtracking?  Maybe this topic deserves a bit more explanation somewhere (and not just in the Ltac chapter).

Here's what I have now:

If not specified, the tactic tries all the constructors,
   which can result in more than one match (e.g. for `\\/`).

Reply:
>  Any feel for how many tactics can produce multiple results with backtracking?

constructor, multimatch, +, typeclasses eauto.  `;` will preserve multiple successes but not introduce new ones.  I think this might be it?

Reply:
I think the solution is to write that `constructor` may produce more than one success with "success" a glossary term (linking back to the Ltac chapter for the readers that want to understand what "producing more than one success" means).